eth/eventhandler/event_handler_test_utils.go

```diff
@@ -0,0 +1,299 @@
+
+	msk, pubk := blskeygen.GenBLSKeyPair()
+	secVec := msk.GetMasterSecretKey(int(sharesCount))
+	pubks := bls.GetMasterPublicKey(secVec)
+	splitKeys, err := threshold.Create(msk.Serialize(), sharesCount-1, sharesCount)
+	if err != nil {
+		return nil, err
+	}
+
+	num := uint64(len(ops))
+	validatorData.operatorsShares = make([]*testShare, num)
+
+	// derive a `hareCount` number of shares
+	for i := uint64(1); i <= num; i++ {
+		validatorData.operatorsShares[i-1] = &testShare{
+			opId: i,
+			sec:  splitKeys[i],
+			pub:  splitKeys[i].GetPublicKey(),
+		}
+	}
+
+	validatorData.masterKey = msk
+	validatorData.masterPubKey = pubk
+	validatorData.masterPublicKeys = pubks
+
+	return validatorData, nil
+}
+
+func createOperators(num uint64, idOffset uint64) ([]*testOperator, error) {
+	testops := make([]*testOperator, num)
+
+	for i := uint64(1); i <= num; i++ {
+		pb, sk, err := rsaencryption.GenerateKeys()
+		if err != nil {
+			return nil, err
+		}
+		testops[i-1] = &testOperator{
+			id:   idOffset + i,
+			pub:  pb,
+			priv: sk,
+		}
+	}
+
+	return testops, nil
+}
+
+func generateSharesData(validatorData *testValidatorData, operators []*testOperator, owner ethcommon.Address, nonce int) ([]byte, error) {
+	var pubkeys []byte
+	var encryptedShares []byte
+
+	for i, op := range operators {
+		rsakey, err := rsaencryption.ConvertPemToPublicKey(op.pub)
+		if err != nil {
+			return nil, fmt.Errorf("cant convert publickey: %w", err)
+		}
+
+		rawshare := validatorData.operatorsShares[i].sec.SerializeToHexStr()
+		ciphertext, err := rsa.EncryptPKCS1v15(rand.Reader, rsakey, []byte(rawshare))
+		if err != nil {
+			return nil, errors.New("cant encrypt share")
+		}
+
+		rsapriv, err := rsaencryption.ConvertPemToPrivateKey(string(op.priv))
+		if err != nil {
+			return nil, err
+		}
+
+		// check that we encrypt right
+		shareSecret := &bls.SecretKey{}
+		decryptedSharePrivateKey, err := rsaencryption.DecodeKey(rsapriv, ciphertext)
+		if err != nil {
+			return nil, err
+		}
+		if err = shareSecret.SetHexString(string(decryptedSharePrivateKey)); err != nil {
+			return nil, err
+		}
+
+		pubkeys = append(pubkeys, validatorData.operatorsShares[i].pub.Serialize()...)
+		encryptedShares = append(encryptedShares, ciphertext...)
+
+	}
+
+	tosign := fmt.Sprintf("%s:%d", owner.String(), nonce)
+	msghash := crypto.Keccak256([]byte(tosign))
+	signed := validatorData.masterKey.Sign(string(msghash))
+	sig := signed.Serialize()
+
+	if !signed.VerifyByte(validatorData.masterPubKey, msghash) {
+		return nil, errors.New("couldn't sign correctly")
+	}
+
+	sharesData := append(pubkeys, encryptedShares...)
+	sharesDataSigned := append(sig, sharesData...)
+
+	return sharesDataSigned, nil
+}
+
+func setupEventHandler(
+	t *testing.T,
+	ctx context.Context,
+	logger *zap.Logger,
+	operator *testOperator,
+	ownerAddress *ethcommon.Address,
+	useMockCtrl bool,
+) (*EventHandler, *mocks.MockController, error) {
+	db, err := kv.NewInMemory(logger, basedb.Options{
```

Comment:
if this is a copy in some way of what used in `cli/operator/node.go`. I think it'll be best to extract `setupEventHandler` to a function that receives most of what it needs (for example a validator controller interface that could be either a mock or actual validator controller instead of useMock bool) and then try to reuse code between `node.go` and the e2e tests, this will make sure we're testing the same functionality used even if code changes in the future.